src/api-service/__app__/onefuzzlib/agent_events.py

```diff
@@ -75,7 +76,9 @@ def on_state_update(
         # they send 'init' with reimage_requested, it's because the node was reimaged
         # successfully.
         node.reimage_requested = False
+        node.initialized_at = datetime.datetime.utcnow()
```

Comment:
Should this be:
```suggestion
        node.initialized_at = datetime.datetime.now(datetime.timezone.utc)
```
The difference (naive vs. TZ-aware) is described here: https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow

Note that our queries which operate on this field use the TZ-aware form.

Reply:
Hm I see them used fairly interchangeably in the code. The queries that we use to check the timer are all datetime.datetime.utcnow(), as well.

Reply:
> Hm I see them used fairly interchangeably in the code. 

That doesn't mean they should be 😄. Take a look at #851.

Example of mixing going awry:
```python
>>> import datetime as dt

# "naive" (not TZ-aware)
>>> now = dt.datetime.utcnow()

# The same time, but TZ-aware
>>> tznow = dt.datetime(2022, 1, 26, 22, 47, 1, 24174, tzinfo=dt.timezone.utc)

# These times are the "same" in many ways
>>> now
datetime.datetime(2022, 1, 26, 22, 47, 1, 24174)
>>> tznow
datetime.datetime(2022, 1, 26, 22, 47, 1, 24174, tzinfo=datetime.timezone.utc)
>>> now.timestamp()
1643237221.024174
>>> tznow.timestamp()
1643237221.024174

# But NOT the same in other ways
>>> now == tznow
False
>>> now <= tznow
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't compare offset-naive and offset-aware datetimes
```

> The queries that we use to check the timer are all datetime.datetime.utcnow(), as well.

What about [this?](https://github.com/microsoft/onefuzz/pull/1616/files#diff-a2210def4c65168b052d6da50ae48f32979d2cc6a5bf02f1fcfd2d510f16e7ebR372)

Reply:
Okay wow I see the issue and yes it looks like the query above uses the appropriate format. Did we ever fully address bug #881? That could be a good follow up to this PR.

Reply:
No, #851 hasn't been addressed yet. I think we do probably want to clean that up to avoid confusing and frustrating errors in the future.

Reply:
Sounds good. I'll make a note of it for our future sync.